From 388c72f416e1b97688d9d229483024c0d91a9c00 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 7 Mar 2019 14:24:50 +0530
Subject: [PATCH 133/215] spi: use gpio_set_value_cansleep for setting
 chipselect GPIO

Sleeping is safe inside spi_transfer_one_message, and some GPIO chips
need to sleep for setting values

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---
 drivers/spi/spi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index c2e85e23d538..4d8cce66f2e8 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -700,7 +700,7 @@ static void spi_set_cs(struct spi_device *spi, bool enable)
 		enable = !enable;
 
 	if (gpio_is_valid(spi->cs_gpio))
-		gpio_set_value(spi->cs_gpio, !enable);
+		gpio_set_value_cansleep(spi->cs_gpio, !enable);
 	else if (spi->master->set_cs)
 		spi->master->set_cs(spi, !enable);
 }
-- 
2.11.0

